activerecord/lib/active_record/railtie.rb

```diff
@@ -357,11 +357,17 @@ class Railtie < Rails::Railtie # :nodoc:
     end
 
     initializer "active_record_encryption.configuration" do |app|
-      ActiveRecord::Encryption.configure \
-         primary_key: app.credentials.dig(:active_record_encryption, :primary_key),
-         deterministic_key: app.credentials.dig(:active_record_encryption, :deterministic_key),
-         key_derivation_salt: app.credentials.dig(:active_record_encryption, :key_derivation_salt),
-         **config.active_record.encryption
+      auto_filtered_parameters = ActiveRecord::Encryption::AutoFilteredParameters.new(app)
+
+      config.after_initialize do |app|
```

Comment:
@rafaelfranca I went with using an `after_initialize` at the end because, even if it's discouraged, it can happen that apps include code that loads ActiveRecord before a given initializer runs, which was resulting in situations where AR encryption got initialized before certain properties had been processed.

Reply:
I've created an issue and PR following that change:

* https://github.com/rails/rails/issues/50604
* https://github.com/rails/rails/pull/50606

I believe this change breaks compatibility with eager loading mode, in which AR models are loaded before the `after_initialize` hook has a chance to set up ARE configs. As a result, ARE's `encrypts` can't access configs that should be already ready but are not.

I'd like to hear your views on this issue.

Thanks.